frappe/core/doctype/audit_trail/audit_trail.js

```diff
@@ -2,6 +2,17 @@
 // For license information, please see license.txt
 
 frappe.ui.form.on("Audit Trail", {
+	onload(frm) {
+		let prev_route = frappe.get_prev_route();
+		if (prev_route.length > 2 && prev_route[0] == "Form" && prev_route[1] != "Audit Trail") {
+			frm.doc.doctype_name = prev_route[1];
+			frm.doc.document = prev_route[2];
+			frm.doc.start_date = "";
+			frm.doc.end_date = "";
+		}
+		frm.events.get_audit_trail_for_document(frm);
+	},
```

Comment:
You might have to move this to refresh with some additional checks. Currently, once you load audit trail, then go back and open audit trail from different transaction record, filters are not updated.

Also, it seems like a bug that single form does not load route_options. [created an issue](https://github.com/frappe/frappe/issues/23061)